src/apps/frontend/pages/tasks/task-modal.tsx

```diff
@@ -0,0 +1,127 @@
+import { FormikProps } from 'formik';
+import React, { useEffect } from 'react';
+
+import {
+  Button,
+  FormControl,
+  Input,
+  VerticalStackLayout,
+} from '../../components';
+import TextArea from '../../components/input/text-area';
+import Modal from '../../components/modal';
+import { AsyncError } from '../../types';
+import { ButtonKind, ButtonType } from '../../types/button';
+import { Task, TaskOperationType } from '../../types/task';
+
+import useTaskForm from './tasks-form.hook';
+
+interface TaskModalProps {
+  btnText: string;
+  formik: FormikProps<Task>;
+  isModalOpen: boolean;
+  onError?: (error: AsyncError) => void;
+  onSuccess?: () => void;
+  setIsModalOpen: (open: boolean) => void;
+  setFieldValue: (
+    formik: FormikProps<Task>,
+    field: string,
+    value: string,
+  ) => void;
+  task?: Task;
+  taskOperationType: string;
+}
+
+const TaskModal: React.FC<TaskModalProps> = ({
+  btnText,
+  formik,
+  isModalOpen,
+  onError,
+  onSuccess,
+  setFieldValue,
+  setIsModalOpen,
+  task,
+  taskOperationType,
+}) => {
+  const { isAddTaskLoading } = useTaskForm({ onSuccess, onError });
+
+  useEffect(() => {
+    if (taskOperationType === TaskOperationType.EDIT.toString() && task) {
+      setFieldValue(formik, 'title', task.title);
+      setFieldValue(formik, 'id', task.id);
+      setFieldValue(formik, 'description', task.description);
```

Comment:
we setting this on edit campaign modal right, in that case we can simply set these fields in the method which is toggling the modal(opening the modal) which is `handleTaskOperation` method in `TaskSection` component in this scenario. There will be no need of useEffect if we do this